tests/wpt/mozilla/tests/mozilla/iterator_invalidation.html

```diff
@@ -0,0 +1,18 @@
+<!doctype html>
+<meta charset="utf-8">
+<title>Should not crash when an iterator is invalidated</title>
+<html>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+<body>
+</body>
+<script>
+    test(function() {
+        var params = new URLSearchParams("foo=bar&baz=qux");
+        params.forEach((p) => {
+           params.delete("baz");
+        })
```

Comment:
Let's push p to an array, and assert the array's contents so it's more than just a "doesn't crash" test.